From e642e6844bd4e6da28efab377f1a94e0eb9a3532 Mon Sep 17 00:00:00 2001
From: Kuldeep Singh <kuldeep.singh@nxp.com>
Date: Fri, 3 Jan 2020 14:47:59 +0530
Subject: [PATCH] LF-20-1 arm64: dts: ls1012ardb: Update qspi node dts
 properties

Update rx and tx bus-width to 1.
Use compatibles as "jedec,spi-nor" to probe flash without displaying warning:
found s25fs512s, expected m25p80

Remove property 'big-endian' as it is not used by new driver anymore.
Also, update dtsi compatibles to use "fsl,ls1021a-qspi".

Signed-off-by: Kuldeep Singh <kuldeep.singh@nxp.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1012a-rdb.dts | 8 ++++----
 arch/arm64/boot/dts/freescale/fsl-ls1012a.dtsi    | 3 +--
 2 files changed, 5 insertions(+), 6 deletions(-)

--- a/arch/arm64/boot/dts/freescale/fsl-ls1012a-rdb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1012a-rdb.dts
@@ -88,15 +88,15 @@
 
 &qspi {
 	status = "okay";
+
 	qflash0: s25fs512s@0 {
-		compatible = "spansion,m25p80";
+		compatible = "jedec,spi-nor";
 		#address-cells = <1>;
 		#size-cells = <1>;
 		spi-max-frequency = <20000000>;
-		m25p,fast-read;
 		reg = <0>;
-		spi-rx-bus-width = <2>;
-		spi-tx-bus-width = <2>;
+		spi-rx-bus-width = <1>;
+		spi-tx-bus-width = <1>;
 	};
 
 };
--- a/arch/arm64/boot/dts/freescale/fsl-ls1012a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1012a.dtsi
@@ -386,7 +386,7 @@
 		};
 
 		qspi: spi@1550000 {
-			compatible = "fsl,ls1012a-qspi", "fsl,ls1021a-qspi";
+			compatible = "fsl,ls1021a-qspi";
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <0x0 0x1550000 0x0 0x10000>,
@@ -395,7 +395,6 @@
 			interrupts = <0 99 IRQ_TYPE_LEVEL_HIGH>;
 			clock-names = "qspi_en", "qspi";
 			clocks = <&clockgen 4 0>, <&clockgen 4 0>;
-			big-endian;
 			status = "disabled";
 		};
 
